File: lib/ratings.ts
// lib/ratings.ts
import { supabase } from './supabase';

/** rating: 1..5、nullで削除 */
export async function upsertRating(animeId: string, rating: number | null) {
  const { error } = await supabase.rpc('upsert_rating', {
    p_anime_id: animeId,
    p_rating: rating
  });
  if (error) throw error;
}

File: lib/lib/ratings.dart
// lib/lib/ratings.dart
import 'package:supabase_flutter/supabase_flutter.dart';
import '../bootstrap_supabase.dart';

Future<void> saveRating({
  required String animeId,
  required int rating, // 1..5
}) async {
  await supabase.rpc('upsert_rating', params: {
    'p_anime_id': animeId,
    'p_rating': rating,
  });
}

// 削除（NULL 指定）
Future<void> deleteRating(String animeId) async {
  await supabase.rpc('upsert_rating', params: {
    'p_anime_id': animeId,
    'p_rating': null,
  });
}